### ISSUE TITLE
Missing geom_curve and units for annotated arrow

### ISSUE THREAD

**Issue by alexhallam** [has2k1/plotnine#535]
I love plotnine. Just ran into an issue where I can't annotate a curved arrow on a plot. There are at least two problems. 

1. No `geom_curve` is implemented
2. No `unit()` is defined

```
p = (
    ggplot(df, aes(x="x", y="y"))
    + geom_point()
    + annotate("text", x = .8, y = .9, label = "Wat!?")
    + annotate(geom = "curve", x = .8, y = .9,xend=.9,yend=.78, curvature=-0.3,arrow=arrow(length=unit(2,"mm")))
    + theme_minimal()
)
```

**Comment by has2k1** [comment]
Yes `geom_curve` is missing. Though, I do not think `unit` is possible given the current backend.

**Comment by W-L** [comment]
I second the original poster, love plotnine! Is `geom_curve` on the roadmap by any chance? Or maybe you could give some pointers where to dig around to contribute this, if there is no fundamental reason that this geom could be implemented?
